invenio_records_resources/services/records/service.py

```diff
@@ -552,3 +558,37 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def reindex_latest_first(
+        self, identity, search_preference=None, extra_filter=None, uow=None
+    ):
+        """Reindexes records matching the query filter, prioritizing latest versions.
+
+        Records will be retrieved from search engine and current versions re-indexed first.
+        The latest version of a record is reindexed twice, but we trade that off for the guarantee that it also gets re-indexed first.
+        """
+        self.require_permission(identity, "manage")
+
+        # Create a search instance with the given filters. We avoid the overhead of executing params interpreters (e.g. aggregations)
+        search = self.create_search(
+            identity,
+            self.record_cls,
+            self.config.search,
+            permission_action="read",
+            extra_filter=extra_filter,
+            preference=search_preference,
+        )
+
+        # Search only latest versions and register them for indexing
+        latest_versions = search.filter("term", **{"versions.is_latest": True}).scan()
```

Comment:
Do you want to be extra safe and handle any `NoResultFound` exception? Maybe simply `continue` in such cases?